keylime/src/structures/capabilities_negotiation.rs

```diff
@@ -0,0 +1,353 @@
+use serde::{Deserialize, Serialize};
+
+// Define the structure for the AttestationRequest:
+#[derive(Serialize, Deserialize, Debug)]
+pub struct AttestationRequest {
+    #[serde(rename(serialize = "data", deserialize = "data"))]
+    pub data: RequestData,
+}
+#[derive(Serialize, Deserialize, Debug)]
+pub struct RequestData {
+    #[serde(rename(serialize = "type", deserialize = "type"))]
+    pub type_: String,
+    pub attributes: Attributes,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Attributes {
+    pub evidence_supported: Vec<EvidenceSupported>,
+    pub boot_time: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct EvidenceSupported {
+    pub evidence_class: String,
+    pub evidence_type: String,
+    pub agent_capabilities: AgentCapabilities,
+    pub version: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct AgentCapabilities {
```

Comment:
I think this should be renamed: these fields are about the TPM capabilities, and not the agent.

Reply:
I have no problem in changing name (indeed, it is updated in the PR) but ... are you proposing this change also to the OpenAPI / JSON? My point is ... should we change information detailed [here](https://gist.github.com/stringlytyped/ab76c6c83e4630be9446576fbeb7534e) ?

Reply:
We should discuss it with HPE guys, nothing is set in stone. Maybe we can keep it as it is now to unblock the rest of the implementation (sorry for making you change it).